test/DynamicExpresso.UnitTest/LiteralsTest.cs

```diff
@@ -75,6 +75,217 @@ public void Numeric_Literals()
 			Assert.AreEqual(-.022M, target.Eval("-.022M"));
 		}
 
+		[Test]
+		public void Numeric_Literals_DefaultLong()
+		{
+			var target = new Interpreter();
+			
+			target.SetDefaultNumberType(DefaultNumberType.Long);
+
+			Assert.IsInstanceOf(typeof(System.Int64), target.Eval("45"));
+
+			Assert.AreEqual(0L, target.Eval("0"));
+			Assert.AreEqual(0.0, target.Eval("0.0"));
+			Assert.AreEqual(45L, target.Eval("45"));
+			Assert.AreEqual(23423423423434L, target.Eval("23423423423434"));
+			Assert.AreEqual(45.5, target.Eval("45.5"));
+			Assert.AreEqual(-0.5, target.Eval("-0.5"));
+			Assert.AreEqual(.2, target.Eval(".2"));
+			Assert.AreEqual(-.2, target.Eval("-.2"));
+			Assert.AreEqual(+.2, target.Eval("+.2"));
+			Assert.AreEqual(.02, target.Eval(".02"));
+			Assert.AreEqual(-.02, target.Eval("-.02"));
+			Assert.AreEqual(+.02, target.Eval("+.02"));
+			Assert.AreEqual(.20, target.Eval(".20"));
+			Assert.AreEqual(-.20, target.Eval("-.20"));
+			Assert.AreEqual(+.20, target.Eval("+.20"));
+			Assert.AreEqual(.201, target.Eval(".201"));
+			Assert.AreEqual(-.201, target.Eval("-.201"));
+			Assert.AreEqual(+.201, target.Eval("+.201"));
+			Assert.AreEqual(2e+201, target.Eval("2e+201"));
+			Assert.AreEqual(2e+20, target.Eval("2e+20"));
+
+			// f suffix (single)
+			Assert.AreEqual(4f, target.Eval("4f"));
+			Assert.AreEqual(45F, target.Eval("45F"));
+			Assert.AreEqual(45.8f, target.Eval("45.8f"));
+			Assert.AreEqual(45.8F, target.Eval("45.8F"));
+			Assert.AreEqual(45.8F, target.Eval(" 45.8F "));
+			Assert.AreEqual(.2f, target.Eval(".2f"));
+			Assert.AreEqual(.2F, target.Eval(".2F"));
+			Assert.AreEqual(-.2f, target.Eval("-.2f"));
+			Assert.AreEqual(-.2F, target.Eval("-.2F"));
+
+			// m suffix (decimal)
+			Assert.AreEqual(5M, target.Eval("5M"));
+			Assert.AreEqual(254m, target.Eval("254m"));
+			Assert.AreEqual(45.232M, target.Eval("45.232M"));
+			Assert.AreEqual(45.232m, target.Eval("45.232m"));
+			Assert.AreEqual(.022M, target.Eval(".022M"));
+			Assert.AreEqual(.022m, target.Eval(".022m"));
+			Assert.AreEqual(-.022m, target.Eval("-.022m"));
+			Assert.AreEqual(-.022M, target.Eval("-.022M"));
+		}
```

Comment:
Could you also add tests with the `u` suffix? For example:

`Assert.AreEqual(45L, target.Eval("45u"));` 

and also tests with the minus unary operator, such as:

`Assert.AreEqual(-45L, target.Eval("-45u"));`

Reply:
I have fixed